modules/apps/data-engine/data-engine-rest-impl/src/main/java/com/liferay/data/engine/rest/internal/resource/v2_0/DataDefinitionResourceImpl.java

```diff
@@ -461,16 +409,52 @@ public DataDefinition putDataDefinition(
 			_ddmStructureLocalService.getDDMStructure(dataDefinitionId),
 			ActionKeys.UPDATE);
 
+		return _addDataDefinitionFieldLinks(
```

Comment:
I saw that you moved `_putDataDefinition` to the `_addDataDefinitionFieldLinks` method, but the code is a bit strange because it has a put method that doesn't call another method called `_putDataDefinition`. There's something wrong with these names.
Checking the old code, `_addDataDefinitionFieldLinks` seems to be an additional step. So if possible, it would be nice to have one call to `_putDataDefinition` and an additional one to `_addDataDefinitionFieldLinks`, or wrap `_addDataDefinitionFieldLinks` inside `_putDataDefinition`.